### ISSUE TITLE
Issues on training SSD to only look a subset of PASCAL VOC's full category

### ISSUE THREAD

**Issue by stoneyang** [weiliu89/caffe#177]
Hi, @weiliu89 

SSD's really a excellent detector, trained on voc dataset. 
I'm curious about to make it just look at some classes of objects, eg, birds, to realize an interesting tools to find those flying creatures in the wild. So I changed the `labelmap_voc.prototxt`, say, just removed the rest classes, leaving only birds and background, tokens are 0 for background, and 1 for bird. And changing `num_classes` in `ssd_pascal_500.py` from 21 to 2. Training gives me `nan` after about 2k iterations. 

Is there something wrong with my configurations? I'm stucked with this problem the whole weekend, any appreciation will be appreciated.

**Comment by weiliu89** [comment]
You have to make sure that the annotated_data_layer produces the correct image with correct ground truth annotations. You can try ssd_pascal.ipynb in examples/. Besides, since you only have 2 classes, there are likely fewer matched boxes in a batch and thus probably you want to lower your learning rate.

**Comment by stoneyang** [comment]
Thanks for your suggestions. 

I've lowered the learning rate , and relaunched the training task several hours ago. Fortunately, it lives after 2k iterations. But, I didn't change the `max_iter` and `stepsize`, say 60k and 40k, respectively. 

Hope this run will stay alive until then. And I will return to report the results.

**Comment by yunhunJang** [comment]
Hi, @stoneyang
Could you explain where should I modify to get ```labelmap_voc.prototxt``` that only contains the subset of the class? I just modified the ```labelmap_voc.prototxt```, but it gives me error. I think it might be caused by the reason that lmdb file contains all 21 classes. So, I'm guessing ```./scripts/create_annoset.py``` is the code to modify.. Is it correct? How can I change it?

Thanks

**Comment by stoneyang** [comment]
Hi, @yunhunJang
 
As stated in my original post, I modified the original `labelmap_voc.prototxt`, i.e., remove uninterested labels and reassign the interested ones from `1` to the last. 

The database should be reconstructed accordingly so that there are only interested labels and their images are collected. To make the new database, you could either collect your own data and make the annotations, or, if your interested labels are the subset of VOC 20 classes, just filter the original VOC dataset. After that, you may finetune @weiliu89 's original SSD model or train your own on the new database. Note that you should change `num_classes` [here](https://github.com/weiliu89/caffe/blob/ssd/examples/ssd/ssd_pascal.py#L266) to be the new number of classes.
